src/huntsman/pocs/camera/pyro/service.py

```diff
@@ -115,3 +130,32 @@ def event_is_set(self, event_type):
 
     def event_wait(self, event_type, timeout):
         return self._get_event(event_type).wait(timeout)
+
+    def _get_camera_config(self, device_name=None):
+        """
+        Retrieve the instance-specific camera config from the config server.
+        Args:
+            device_name (str, optional): The string used to query the config server for the
+                instance-specific camera config. If None or not found will attempt to use
+                the device's own IP address.
+        Returns:
+            dict: The camera config dictionary.
+        """
+        device_configs = self.config.get("cameras")["devices"]
+
+        # Try and use the device name to select the device config
+        if device_name is not None:
+            for config in device_configs:
+                if config["name"] == device_name:
+                    self.logger.debug(f"Found camera config by name for {device_name}.")
+                    return config
+            self.logger.debug(f"Unable to find config entry for {device_name}.")
+
+        # If no match for device name, attempt to use IP address
+        ip_address = get_own_ip()
+        self.logger.debug(f"Querying for camera config with identifier: {ip_address}.")
+        for config in device_configs:
+            if config["name"] == ip_address:
+                return config
```

Comment:
@wtgee  The problem with this is that the device name is not the key, but rather a property of the device. This is by design in order to be consistent with `pocs`, which expects a list of devices rather than a dict.